app/controllers/events_controller.rb

```diff
@@ -0,0 +1,75 @@
+class EventsController < ApplicationController
+  before_action :set_event, only: [:show, :edit, :update, :destroy]
+  before_action :authorize_project_member
+  before_action :authorize_event_author, only: [:update, :destroy]
+
+  def index
+    @project = Project.find(params[:project_id])
```

Comment:
In the situation when there are no events created within project, I still want to have the link to creating new event associated to the project. I wouldn't have any event to get project_id from so I need to pass it separately. I fetch and pass only single project instance to the view.